lib/matplotlib/patches.py

```diff
@@ -138,9 +138,18 @@ def contains(self, mouseevent, radius=None):
         if callable(self._contains):
             return self._contains(self, mouseevent)
         radius = self._process_radius(radius)
-        inside = self.get_path().contains_point(
-            (mouseevent.x, mouseevent.y), self.get_transform(), radius)
-        return inside, {}
+        codes = self.get_path().codes
+        vertices = self.get_path().vertices
+        # if the current path is concatenated by multiple sub paths.
+        # get the indexes of the starting code(MOVETO) of all sub paths
+        idxs, = np.where(codes == 1)
+        # Don't split before the first MOVETO.
+        idxs = idxs[1:]
+        return any(
+            subpath.contains_point(
+                (mouseevent.x, mouseevent.y), self.get_transform(), radius)
+            for subpath in map(
+                Path, np.split(vertices, idxs), np.split(codes, idxs))), {}
```

Comment:
I think I finally see what this is doing, but,  can this be deconvolved to ease in maintenance and understanding?  Other devs are free to approve if they are OK with this, and its just my python ignorance, but its not the clearest return statement in the code base.

Reply:
Could you reformat this to

~~~
        subpaths = map(Path, np.split(vertices, idxs), np.split(codes, idxs)))
        contains = any(
            subpath.contains_point(
                (mouseevent.x, mouseevent.y), self.get_transform(), radius)
            for subpath in subpaths))
	return contains, {}
~~~

Reply:
I think it is OK to leave as-is.